ecllibrary/std/system/Store.ecl

```diff
@@ -0,0 +1,573 @@
+ *                          to an empty string
+ * @param   espURL          The full URL for accessing the esp process running
+ *                          on the HPCC Systems cluster (this is typically the
+ *                          same URL as used to access ECL Watch); set to an
+ *                          empty string to use the URL of the current esp
+ *                          process; OPTIONAL, defaults to an empty string
+ *
+ * @return  A reference to the module, correctly initialized with the
+ *          given access parameters.
+ */
+EXPORT Store(STRING username = '',
+             STRING userPW = '',
+             STRING espURL = '') := MODULE
+
+    SHARED ENCODED_CREDENTIALS := IF
+        (
+            TRIM(username, ALL) != '',
+            'Basic ' + Std.Str.EncodeBase64((DATA)(TRIM(username, ALL) + ':' + TRIM(userPW, LEFT, RIGHT))),
+            ''
+        );
+
+    // The URL that will be used by all SOAPCALL invocations
+    TRIMMED_URL := TRIM(espURL, ALL);
+    SHARED MY_ESP_URL := IF(TRIMMED_URL != '', TRIMMED_URL, Std.File.GetEspURL()) + '/WsStore/?ver_=1.02';
+
+    /**
+     * Helper for function for setting the has_exception field within a
+     * response record.
+     *
+     * @param   rec     The single RECORD containing a WsStore response
+     *
+     * @return  The same record with the has_exceptions field set to TRUE
+     *          or FALSE based on examining the exception list.
+     */
+    SHARED UpdateForExceptions(res) := FUNCTIONMACRO
+        RETURN PROJECT
+            (
+                res,
+                TRANSFORM
+                    (
+                        RECORDOF(LEFT),
+                        SELF.has_exceptions := EXISTS(LEFT.exceptions.exceptions),
+                        SELF := LEFT
+                    )
+            );
+    ENDMACRO;
+
+    //--------------------------------------------------------------------------
+    // Exported record definitions; all used as results of SOAPCALL invocations
+    //--------------------------------------------------------------------------
+
+    EXPORT ExceptionLayout := RECORD
+        STRING                          code            {XPATH('Code')};
+        STRING                          audience        {XPATH('Audience')};
+        STRING                          source          {XPATH('Source')};
+        STRING                          message         {XPATH('Message')};
+    END;
+
+    EXPORT ExceptionListLayout := RECORD
+        STRING                          source          {XPATH('Source')};
+        DATASET(ExceptionLayout)        exceptions      {XPATH('Exception')};
+    END;
+
+    EXPORT StoreInfoRec := RECORD
+        STRING                          store_name      {XPATH('Name')};
+        STRING                          description     {XPATH('Description')};
+        STRING                          owner           {XPATH('Owner')};
+        STRING                          create_time     {XPATH('CreateTime')};
```

Comment:
FYI The storeinfo has a couple of new response fields in 7.6.0

`<MaxValSize>1024</MaxValSize>    //each entry in the value-set is restricted by this size value
<IsDefault>1</IsDefault>`